plugins/queue/scheduler.js

```diff
@@ -0,0 +1,614 @@
+            // eslint-disable-next-line max-len
+            if (err && err.message !== 'Job already enqueued at this time with same arguments') {
+                shouldRetry = true;
+            }
+        }
+        if (!shouldRetry) {
+            return Promise.resolve();
+        }
+        try {
+            await executor.queue.enqueueAt(next, executor.periodicBuildQueue, 'startDelayed', [{ jobId: job.id }]);
+        } catch (err) {
+            logger.error(`failed to add to delayed queue for job ${job.id}: ${err}`);
+        }
+    }
+
+    return Promise.resolve();
+}
+
+/**
+ * Starts a new build in an executor
+ * @async  start
+ * @param  {Object} config               Configuration
+ * @param  {Object} [config.annotations] Optional key/value object
+ * @param  {Number} [config.eventId]     Optional eventID that this build belongs to
+ * @param  {String} config.build         Build object
+ * @param  {Array}  config.blockedBy     Array of job IDs that this job is blocked by. Always blockedby itself
+ * @param  {String} config.causeMessage  Reason the event is run
+ * @param  {Array}  config.freezeWindows Array of cron expressions that this job cannot run during
+ * @param  {String} config.apiUri        Screwdriver's API
+ * @param  {String} config.jobId         JobID that this build belongs to
+ * @param  {String} config.jobName       Name of job that this build belongs to
+ * @param  {String} config.jobState      ENABLED/DISABLED
+ * @param  {String} config.jobArchived   Boolean value of whether job is archived
+ * @param  {String} config.buildId       Unique ID for a build
+ * @param  {Object} config.pipeline      Pipeline of the job
+ * @param  {Fn}     config.tokenGen      Function to generate JWT from username, scope and scmContext
+ * @param  {String} config.container     Container for the build to run in
+ * @param  {String} config.token         JWT to act on behalf of the build
+ * @return {Promise}
+ */
+async function start(executor, config) {
+    await executor.connect();
+    const {
+        build,
+        buildId,
+        causeMessage,
+        jobId,
+        jobState,
+        jobArchived,
+        blockedBy,
+        freezeWindows,
+        token,
+        apiUri
+    } = config;
+    const forceStart = /\[(force start)\]/.test(causeMessage);
+
+    delete config.build;
+    delete config.causeMessage;
+
+    await stopFrozen(executor, {
+        jobId
+    });
+
+    // Skip if job is disabled or archived
+    if (jobState === 'DISABLED' || jobArchived === true) {
+        return Promise.resolve();
+    }
+
+    const currentTime = new Date();
+    const origTime = new Date(currentTime.getTime());
+
+    timeOutOfWindows(freezeWindows, currentTime);
+
+    let enq;
+
+    // Check freeze window
+    if (currentTime.getTime() > origTime.getTime() && !forceStart) {
+        await helper
+            .updateBuild(
```

Comment:
error handling is wtong https://github.com/screwdriver-cd/queue-service/blob/master/plugins/helper.js#L209-L211